stl/inc/charconv

```diff
@@ -1225,52 +1225,116 @@ _NODISCARD inline bool _Should_round_up(
 // The caller must pass true for _Has_zero_tail if all discarded bits were zeroes.
 _NODISCARD inline uint64_t _Right_shift_with_rounding(
     const uint64_t _Value, const uint32_t _Shift, const bool _Has_zero_tail) noexcept {
-    // If we'd need to shift further than it is possible to shift, the answer is always zero:
     if (_Shift >= 64) {
-        return 0;
+        if (_Shift == 64) {
+            const uint64_t _Round_bit = _Value >> 63;
+            const auto _Has_tail_bits = static_cast<uint64_t>(!_Has_zero_tail);

```

Comment:
@mnatsuhara noted that if this is indeed a bug, we should add a test case for it. I believe we were unable to actually exercise this codepath with the public interface of `from_chars`, but were equally unable to prove that it can never ever be executed. So, I think it would make sense to directly test this simple internal function, just in case it can be executed. A small set of tests would be reasonable to add to the existing charconv test.